java/configuration/src/main/java/sleeper/configuration/properties/SystemDefinedInstancePropertyImpl.java

```diff
@@ -40,16 +46,52 @@ public String getPropertyName() {
         return propertyName;
     }
 
-    static SystemDefinedInstanceProperty named(String propertyName) {
-        return addToAllList(new SystemDefinedInstancePropertyImpl(propertyName));
+    @Override
+    public String getDescription() {
+        return description;
+    }
+
+    public static Builder builder() {
+        return new Builder();
     }
 
-    private static SystemDefinedInstanceProperty addToAllList(SystemDefinedInstanceProperty property) {
-        ALL.add(property);
-        return property;
+    public static Builder named(String propertyName) {
+        return builder().propertyName(propertyName);
     }
 
-    static List<SystemDefinedInstanceProperty> all() {
+    public static SystemDefinedInstanceProperty get(String propertyName) {
+        return ALL_MAP.get(propertyName);
+    }
+
+    public static List<SystemDefinedInstanceProperty> all() {
         return Collections.unmodifiableList(ALL);
     }
```

Comment:
In most classes where we use a builder we've put the static constructors and methods for getting a builder at the top next to the constructor. I find that helpful as you can see the main code for creating an instance together.